File: benchmark/bit_twiddle/bit_twiddle.go
package main

import "fmt"

func main() {
	iterations := 20000000
	var x uint32 = 2463534242
	var sum uint32 = 0

	for i := 0; i < iterations; i++ {
		x ^= x << 13
		x ^= x >> 17
		x ^= x << 5
		sum += x & 1023
	}

	fmt.Printf("Bit twiddle: %d %d\n", x, sum)
}

File: benchmark/bit_twiddle/bit_twiddle.py
def main():
    iterations = 20000000
    x = 2463534242
    total = 0
    mask = 0xFFFFFFFF

    for _ in range(iterations):
        x = (x ^ (x << 13)) & mask
        x = (x ^ (x >> 17)) & mask
        x = (x ^ (x << 5)) & mask
        total = (total + (x & 1023)) & mask

    print(f"Bit twiddle: {x} {total}")


if __name__ == "__main__":
    main()
